### ISSUE TITLE
update geometries from geonames?

### ISSUE THREAD

**Issue by missinglink** [whosonfirst-data/whosonfirst-data-postalcode-pl#1]
The geometries in this repo are currently all `0,0`.

I had a quick look at it this morning and it looks like we can source point geometries for ~21k of the ~25k postcodes in this repo from Geonames:

```bash
sqlite3 whosonfirst-data-postalcode-pl-latest.db -separator $'\t' $'SELECT id, json_extract(body, \'$.properties."wof:name"\') from geojson' > wof.txt
```

```bash
aria2c http://www.geonames.org/export/zip/PL.zip
```

```bash
join -12 -22 <(sort -k2 wof.txt) <(sort -k2 PL.txt) | head 
00-001 421385723 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-002 421412029 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-003 421395245 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-004 421395247 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-005 421395249 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-006 421395251 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-007 421395253 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-008 421395257 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-009 421395259 PL Warszawa Mazowieckie Warszawa 52.25 21 4
00-010 421395261 PL Warszawa Mazowieckie Warszawa 52.25 21 4
```

(the second column is the WOF ID)

**Comment by missinglink** [comment]
I've not worked with these Geonames postcode files before, my initial impression is that the coordinates appear to be very low precision and duplicated.

Not sure if this is 'normal' although it might indicate an error in the data or how Geonames processes it.

**Comment by stepps00** [comment]
The initial postalcode import (6+ years ago) into Who's On First came via GeoPlanet, which mostly contained geometries with `0,0` coordinates.

GeoNames exists as a [source]() in Who's On First already, so assuming the same license applies to GeoNames' postalcode data, we should be able to import.

@missinglink In previous imports of GeoNames' locality data, I've noticed the same low geometric precision. IMO, this is preferred over the `0,0` coordinates though. Would you be interested in crafting a PR with updates? If not, I can try to spend some time on this later in the week.

**Comment by missinglink** [comment]
Yeah I can do a PR no problem, I'm just holding off waiting for a reply from the OP with their local knowledge of the postcodes and their accuracy before continuing. I'd prefer not to waste time if the GN postcodes are really bad.

My inclination is more toward "nothing is better than wrong" so I'd prefer to have `0,0` than very low precision positions.

The advantage to us is that we can exclude null island geometries easily but don't have a way of excluding low quality/precision geometries.

The problem with having some really good quality geoms and some really low quality geoms is that people always judge the quality of your dataset/product/service by the worst example they can find, and then usually make assumptions that the rest is of the same quality 🤷‍♂️

**Comment by missinglink** [comment]
For example there are multiple postcodes in the GN dataset with the coords `52.25,21`.

Sorry I should have really done my homework before opening this issue but the OP was claiming they were significantly better in GN.
